### PR TITLE
api/utils/ui: List protocol numbers and icmp types

### PR DESCRIPTION
**Pull request by weizhouapache**
### Description

This PR contains the following changes
- adds a new API to list network procotols and details/types/codes, etc
- get network protocols on UI and add dropdowns for procotol numbers and icmp types/codes
- validate icmp types/codes when add network ACL
- etc

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

File: utils/src/main/java/com/cloud/utils/net/NetworkProtocols.java [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+        ProtocolNumbers.add(new ProtocolNumber(139, "HIP", "Host Identity Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(140, "Shim6", "Shim6 Protocol"));
+        ProtocolNumbers.add(new ProtocolNumber(141, "WESP", "Wrapped Encapsulating Security Payload"));
+        ProtocolNumbers.add(new ProtocolNumber(142, "ROHC", "Robust Header Compression"));
+        ProtocolNumbers.add(new ProtocolNumber(143, "Ethernet", "Ethernet"));
+        ProtocolNumbers.add(new ProtocolNumber(144, "AGGFRAG", "AGGFRAG encapsulation payload for ESP"));
+        ProtocolNumbers.add(new ProtocolNumber(145, "NSH", "Network Service Header"));
+    }
+    /**
+     * Different types of ICMP.
```
Comment (DaanHoogland) [inline utils/src/main/java/com/cloud/utils/net/NetworkProtocols.java:193]: is this comment correct? It seems any protocol could be encoded not just ICMP

**Review by DaanHoogland — APPROVED** [review]
code looks good, did a small monkey test on both ui and API but might need more testing mainly in advancedsg zone

File: utils/src/main/java/com/cloud/utils/net/NetworkProtocols.java (hunk 1 above)
Comment (weizhouapache) [reply utils/src/main/java/com/cloud/utils/net/NetworkProtocols.java:193]: thanks @DaanHoogland for the review

the comment needs to be put to line 222 I think. I will make some other changes on the comments

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by kiranchavala** [comment]
@blueorangutan package

**Review by kiranchavala — APPROVED** [review]
LGTM, tested the feature manually, with Isolated networks (firewall rules) , VPC (network acl), and security group rules in advanced zone


The dropdowns ( protocols numbers, icmp type and icmp code )  in the UI is working fine

**Comment by weizhouapache** [comment]
this has been tested by @kiranchavala and @alexandremattioli 

@shwstppr 
ready for merge to main after 4.19.0.0 release

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Pull request merged**